### PR TITLE
Bump dependencies

### PR DESCRIPTION
**Pull request by thofma**

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -24,18 +24,18 @@ UUIDs = "cf7118a7-6976-5b1a-9a39-7adc72f591a4"
...
 
 [compat]
-AbstractAlgebra = "0.30.9"
-AlgebraicSolving = "0.3.0"
+AbstractAlgebra = "0.31.0"
+AlgebraicSolving = "0.3.3"
 DocStringExtensions = "0.8, 0.9"
 GAP = "0.9.4"
-Hecke = "0.18.13"
+Hecke = "0.19.0"
```
Comment (thofma) [inline Project.toml]:
```suggestion
Hecke = "0.19.1"
```

File: Project.toml [hunk 2]
```diff
@@ -24,18 +24,18 @@ UUIDs = "cf7118a7-6976-5b1a-9a39-7adc72f591a4"
...
 
 [compat]
-AbstractAlgebra = "0.30.9"
-AlgebraicSolving = "0.3.0"
+AbstractAlgebra = "0.31.0"
+AlgebraicSolving = "0.3.3"
 DocStringExtensions = "0.8, 0.9"
 GAP = "0.9.4"
-Hecke = "0.18.13"
+Hecke = "0.19.1"
```
Comment (thofma) [inline Project.toml]:
```suggestion
Hecke = "0.19.2"
```

**Comment by lgoettgens** [comment]
This PR is dedicated to collecting all changes necessary to bump the Nemo and Hecke compats to the next minor (breaking) release. Many of them have already been proposed as a PR against master and have been changed to be merged here.

**Comment by thofma** [comment]
Should be good to go. And should be squashed.

**Comment by thofma** [comment]
If there are no objections, I will merge this at the end of the day.

**Review by benlorenz — APPROVED** [review]
LGTM

**Pull request merged**